templates/hsdk-ngc-container/README.md

```diff
@@ -0,0 +1,93 @@
+- [NVIDIA CUDA Toolkit](https://developer.nvidia.com/cuda-downloads)
+
+## DevContainer Setup
+
+1. Download everything in this directory to a folder of your choice.
+2. Open [.devcontainer/Dockerfile](.devcontainer/Dockerfile) and change the `ARG HOLOSCAN_SDK_IMAGE` line to match the [Holoscan SDK container]((https://catalog.ngc.nvidia.com/orgs/nvidia/teams/clara-holoscan/containers/holoscan)) image you want to use.
+3. Start VS Code, press `F1` or `Ctrl+Shift+P`, type to select `Dev Containers: Open Folder in Container...` and select the folder where you stored step 1.
+4. Wait for the Dev Container to start up. Building the container and installing all the required extensions will take a while. (Switch to the `Output` view and select the `Server` option from the dropdown to check the status of extension installations.)
+5. Finally, from the **File** menu, select **Open Workspace from File...** and select `/workspace/.vscode/workspace.code-workspace` to open a pre-configured workspace.
+
+When everything is ready, you should see the following in the VS Code Explorer sidebar.
+
+*Tip: The screenshot below shows a pink clock icon over the Extensions sidebar, which indicates that VS Code is still installing all the required extensions.*
+
+![VS Code Explorer](./static/vscode-explorer.png)
+
+The `My Workspace` section contains all the files that you've downloaded. The `Holoscan SDK` section includes the Holoscan example applications from `~/examples`.
+
+### Directory Structure
+
+```bash
+├── .devcontainer/: contains the Dev Container configuration file and Dockerfile that you've downloaded
+├── .vscode/: contains the VSCode configuration files for debugging
+├── build/: contains the Holoscan sample applications built by CMake
+├── src/: where you may start your development
+├── static/: contains images and other assets used in this README
+├── .code-workspace: pre-configured VSCode workspace described in step 5 above
+└── README.md: this file
+```
+
+## Debugging Applications
+
+
+### Debugging a C++ Application
+
+This section will walk you through the steps to debug the [Hello World](../home/holoscan/examples/hello_world/README.md) application using VS Code.
+
+Click on **Explorer** from the sidebar and expand the **examples** folder. Find and open `hello_world.cpp` under `hello_world/cpp` directory.
+Let's put a breakpoint on this line: `auto app = holoscan::make_application<HelloWorldApp>();` (Feel free to put more breakpoints wherever you want).
+
+Now, let's switch to the **Run and Debug** panel on the sidebar, and then click on the dropdown box as shown below:
+![VS Code Run and Debug](./static/vscode-run-and-debug.png)
+
+Select `(gdb) examples/hello_world/cpp` from the list of available launch configurations.
+
+Hit **F5** on the keyboard or click the green arrow to start debugging. VS Code shall hit the breakpoint and stop as the screenshot shows below:
```

Comment:
I am now copying all .vscode profiles from the NGC container, so each example app now has its launch profile and requires no build since it's now calling the precompiled binaries.